### PR TITLE
Prepare srpm build by adding makefile rules

### PR DESCRIPTION
**Pull request by miyunari**
Hi, this PR adds two rules to the Makefile. :smiley: 
1. `vendor`: to download dependencies (should we add the vendor folder to `.gitignore`?)
2. `archive`: to generate source archives for the collector and its dependencies

As a follow up I would contribute a spec file and prepare a copr repository.

### CODE REVIEW

**Comment by andreasgerstmayr** [comment]
> I assume adding the vendor folder to `.gitignore` might be a good thing.
> 
> wdyt @pavolloffay @andreasgerstmayr ?

+1, I don't think we should store the vendor directory in this repository.

**Comment by miyunari** [comment]
@frzifus @andreasgerstmayr I have put the `vendor` path on the `.gitignore` list. :smiley:

File: Makefile [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 OTELCOL_BUILDER ?= ${OTELCOL_BUILDER_DIR}/ocb
+PROJECT ?= redhat-otel-collector
 
-build: ocb
+build: ocb vendor
 	mkdir -p _build
 	${OTELCOL_BUILDER} --skip-compilation=false --go ${GO} --config manifest.yaml 2>&1 | tee _build/build.log
 
-generate-sources: ocb
+generate-sources: ocb vendor
```
Comment (frzifus) [inline Makefile:11]:
```suggestion
generate-sources: ocb
```

**Comment by frzifus** [comment]
Since vendoring is only relevant for the srpm, I assume we do not need it attached to the build rule.

**Review by frzifus — APPROVED** [review]

**Pull request merged**